File: datasets/rb-keep-code_davinci_001_temp_0.2/HumanEval_46_fib4.rb
# The Fib4 number sequence is a sequence similar to the Fibbonacci sequnece that's defined as follows:
# fib4(0) -> 0
# fib4(1) -> 0
# fib4(2) -> 2
# fib4(3) -> 0
# fib4(n) -> fib4(n-1) + fib4(n-2) + fib4(n-3) + fib4(n-4).
# Please write a function to efficiently compute the n-th element of the fib4 number sequence.  Do not use recursion.
# >>> fib4(5)
# 4
# >>> fib4(6)
# 8
# >>> fib4(7)
# 14
def fib4(n)
  return 0 if n == 0
  return 0 if n == 1
  return 2 if n == 2
  return 0 if n == 3
  return fib4(n-1) + fib4(n-2) + fib4(n-3) + fib4(n-4)
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_fib4
    candidate = method(:fib4)
    assert_equal(4, candidate.call(5))
    assert_equal(28, candidate.call(8))
    assert_equal(104, candidate.call(10))
    assert_equal(386, candidate.call(12))
  end
end

File: datasets/lua-keep-incoder_6b_temp_0.2-3/HumanEval_46_fib4.lua
-- The Fib4 number sequence is a sequence similar to the Fibbonacci sequnece that's defined as follows:
-- fib4(0) -> 0
-- fib4(1) -> 0
-- fib4(2) -> 2
-- fib4(3) -> 0
-- fib4(n) -> fib4(n-1) + fib4(n-2) + fib4(n-3) + fib4(n-4).
-- Please write a function to efficiently compute the n-th element of the fib4 number sequence.  Do not use recursion.
-- >>> fib4(5)
-- 4
-- >>> fib4(6)
-- 8
-- >>> fib4(7)
-- 14
local function fib4(n)
  if n < 0 then
    return 0
  elseif n == 0 then
    return 0
  elseif n == 1 then
    return 1
  elseif n == 2 then
    return 2
  elseif n == 3 then
    return 0
  elseif n == 4 then
    return 0
  else
    return fib4(n - 1) + fib4(n - 2) + fib4(n - 3) + fib4(n - 4)
  end
end


lu = require('luaunit')

function test_humaneval()
local candidate = fib4
    lu.assertEquals(candidate(5), 4)
    lu.assertEquals(candidate(8), 28)
    lu.assertEquals(candidate(10), 104)
    lu.assertEquals(candidate(12), 386)
end

os.exit(lu.LuaUnit.run())